ipn/ipnserver/server_windows.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build windows
+
+package ipnserver
+
+import (
+	"fmt"
+	"net"
+	"runtime"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+	"tailscale.com/util/pidowner"
+)
+
+var (
+	kernel32                        = syscall.NewLazyDLL("kernel32.dll")
+	procGetNamedPipeClientProcessId = kernel32.NewProc("GetNamedPipeClientProcessId")
+)
+
+func getNamedPipeClientProcessId(h windows.Handle) (pid uint32, err error) {
+	r1, _, err := procGetNamedPipeClientProcessId.Call(uintptr(h), uintptr(unsafe.Pointer(&pid)))
+	if r1 > 0 {
+		return pid, nil
+	}
+	return 0, err
+}
+
+// getConnIdentity extracts the identity information from the connection.
+// It expects the passed-in connection to be a localhost TCP connection over
+// named pipes.
+func (s *Server) getConnIdentity(c net.Conn) (ci connIdentity, err error) {
+	ci = connIdentity{Conn: c}
+	h, ok := c.(interface {
+		Handle() windows.Handle
+	})
+	if !ok {
+		return ci, fmt.Errorf("not a windows handle: %T", c)
+	}
+	pid, err := getNamedPipeClientProcessId(h.Handle())
+	if err != nil {
+		return ci, fmt.Errorf("getNamedPipeClientProcessId: %v", err)
+	}
+	ci.Pid = int(pid)
+	uid, err := pidowner.OwnerOfPID(ci.Pid)
+	if err != nil {
+		var hint string
+		if runtime.GOOS == "windows" {
```

Comment:
Is this necessary now that this is in a `_windows.go` file?

Reply:
Done